.pre-commit-hooks.yaml

```diff
@@ -42,9 +42,9 @@
     stages: [commit, push, manual]
 -   id: check-illegal-windows-names
     name: check illegal windows names
-    entry: Illegal windows filenames detected
+    entry: Illegal Windows filenames detected
     language: fail
-    files: '(?i)((^|/)(CON|PRN|AUX|NUL|COM[1-9]|LPT[1-9])(\.|/|$)|:)'
```

Comment:
at the very least the changes to `COM` and likely LPT do not seem correct:

```
C:\Users\Anthony\y>dir
 Volume in drive C has no label.
 Volume Serial Number is A6A6-1A8E

 Directory of C:\Users\Anthony\y

04/27/2024  02:27 PM    <DIR>          .
04/27/2024  02:27 PM    <DIR>          ..
04/27/2024  02:27 PM                 0 com0
04/27/2024  02:27 PM                 0 com¹
               2 File(s)              0 bytes
               2 Dir(s)  239,653,076,992 bytes free
```

Reply:
I've taken another at it. I am actually able to create files named `com0`, `lpt0`, and `clock$`. I was surprisingly also able to create illegal filenames if there was an extension, so `COM1.txt` works perfectly fine for me as well. However, I cannot make files named `com¹` or `lpt¹`. I'm curious how you were actually able to create `com¹`. 

I've seen in [this video](https://youtu.be/s1ct8VPB-4U?si=6PS_BHzFYBkNkKlq&t=751) that creating `COM1.txt` on Windows 10 was illegal and behaves weirdly if you do it anyway. But as I mentioned before, it worked fine for me on Windows 11 23H2. Perhaps different versions of Windows have different restrictions? I can imagine that the Windows devs are slowly getting rid of these restrictions, so maybe `COM1.txt` works in most use cases now, but still not in all. 

So just to be safe, I think it's fine to stick to [Microsoft's own documentation](https://learn.microsoft.com/en-us/windows/win32/fileio/naming-a-file#naming-conventions), even if it is slightly more restrictive than Windows actually is. I will remove `clock$` though since that is not actually listed there and it seems to only affect Windows NT and older.